solver/app/v2/quote.go

```diff
@@ -0,0 +1,83 @@
+package appv2
+
+import (
+	"math/big"
+
+	"github.com/omni-network/omni/lib/errors"
+)
+
+// getQuote returns payment in `depositTkns` required to pay for `expenses`.
+//
+// For now, this is a simple quote that requires a single expense, paid
+// for by an equal amount of an equivalent deposit token. Token equivalence is
+// determined by symbol (ex arbitrum "ETH" is equivalent to optimism "ETH").
+func getQuote(depositTkns []Token, expenses []Payment) ([]Payment, RejectOrErr) {
+	if len(depositTkns) != 1 {
+		return nil, RejectOrErr{
+			Reason: rejectInvalidDeposit,
+			Err:    errors.New("only single deposit token supported"),
+		}
+	}
+
+	if len(expenses) != 1 {
+		return nil, RejectOrErr{
+			Reason: rejectInvalidExpense,
+			Err:    errors.New("only single expense supported"),
+		}
+	}
+
+	expense := expenses[0]
+	depositTkn := depositTkns[0]
+
+	if expense.Token.Symbol != depositTkn.Symbol {
```

Comment:
will add a check here that makes sure a testnet token cannot pay for mainnet token (like holesky ETH for mainnet ETH)

technically this wouldn't be able to happen because the networks would not overlap, so solver would reject for "unsupported dest chain".

but probably still good to check here